### ISSUE TITLE
Fix Elixir 1.3 and other warnings

### ISSUE THREAD

**Issue by joshprice** [seanabrahams/graphql-relay-elixir#19]
Some of these will be fixed by https://github.com/graphql-elixir/graphql/issues/89 but there are quite a number of others.

```
✪ mix test
WARN:  Missing plugins: [pc]
==> esqlite (compile)
Compiled src/esqlite3_nif.erl
Compiled src/esqlite3.erl
Compiling c_src/esqlite3_nif.c
Compiling c_src/queue.c
Compiling c_src/sqlite3.c
c_src/sqlite3.c:9111:26: warning: unused variable 'sqlite3one' [-Wunused-const-variable]
SQLITE_PRIVATE const int sqlite3one = 1;
                         ^
1 warning generated.
==> poolboy (compile)
Compiled src/poolboy_worker.erl
Compiled src/poolboy_sup.erl
Compiled src/poolboy.erl
==> decimal
Compiling 1 file (.ex)
Generated decimal app
==> graphql
Compiling 1 file (.yrl)
Compiling 1 file (.xrl)
Compiling 2 files (.erl)
Compiling 48 files (.ex)
warning: the variable "accumulator" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/lang/ast/type_info_visitor.ex:149

warning: the variable "accumulator" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/lang/ast/type_info_visitor.ex:170

warning: the variable "acc" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/validation/rules/no_fragment_cycles.ex:75

warning: the variable "acc" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/validation/rules/no_fragment_cycles.ex:75

warning: the variable "accumulator" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/validation/rules/fields_on_correct_type.ex:27

warning: the variable "message" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/validation/rules/fields_on_correct_type.ex:91

warning: function GraphQL.Lang.AST.TypeInfo.nullable_type/2 is undefined or private
  lib/graphql/lang/ast/type_info_visitor.ex:126

Generated graphql app
==> poison
Compiling 4 files (.ex)
Generated poison app
==> pipe
Compiling 1 file (.ex)
Generated pipe app
==> ecto
Compiling 73 files (.ex)
warning: function Ecto.Pools.Poolboy.Worker.behaviour_info/1 is undefined or private
  lib/ecto/pools/poolboy/worker.ex:5

warning: function Ecto.Adapters.MySQL.Connection.all/1 is undefined (module Ecto.Adapters.MySQL.Connection is not available)
  lib/ecto/adapters/mysql.ex:85

warning: function Ecto.Adapters.MySQL.Connection.delete/4 is undefined (module Ecto.Adapters.MySQL.Connection is not available)
  lib/ecto/adapters/mysql.ex:85

warning: function Ecto.Adapters.MySQL.Connection.delete_all/1 is undefined (module Ecto.Adapters.MySQL.Connection is not available)
  lib/ecto/adapters/mysql.ex:85

warning: function Ecto.Adapters.MySQL.Connection.execute_ddl/1 is undefined (module Ecto.Adapters.MySQL.Connection is not available)
  lib/ecto/adapters/mysql.ex:85

warning: function Ecto.Adapters.MySQL.Connection.update/5 is undefined (module Ecto.Adapters.MySQL.Connection is not available)
  lib/ecto/adapters/mysql.ex:85

warning: function Ecto.Adapters.MySQL.Connection.update_all/1 is undefined (module Ecto.Adapters.MySQL.Connection is not available)
  lib/ecto/adapters/mysql.ex:85

warning: function Ecto.Adapters.MySQL.Connection.insert/4 is undefined (module Ecto.Adapters.MySQL.Connection is not available)
Found at 2 locations:
  lib/ecto/adapters/mysql.ex:85
  lib/ecto/adapters/mysql.ex:172

warning: function Ecto.Adapters.MySQL.Connection.to_constraints/1 is undefined (module Ecto.Adapters.MySQL.Connection is not available)
  lib/ecto/adapters/mysql.ex:177

warning: function :sbroker.async_ask_r/3 is undefined (module :sbroker is not available)
Found at 2 locations:
  lib/ecto/pools/sojourn_broker/worker.ex:192
  lib/ecto/pools/sojourn_broker/worker.ex:198

warning: function :sbroker.cancel/3 is undefined (module :sbroker is not available)
  lib/ecto/pools/sojourn_broker/worker.ex:260

warning: function :sbroker.await/2 is undefined (module :sbroker is not available)
  lib/ecto/pools/sojourn_broker/worker.ex:261

warning: function :sbroker.ask/2 is undefined (module :sbroker is not available)
  lib/ecto/pools/sojourn_broker.ex:75

warning: function Ecto.Adapters.Postgres.Connection.all/1 is undefined (module Ecto.Adapters.Postgres.Connection is not available)
  lib/ecto/adapters/postgres.ex:56

warning: function Ecto.Adapters.Postgres.Connection.delete/4 is undefined (module Ecto.Adapters.Postgres.Connection is not available)
  lib/ecto/adapters/postgres.ex:56

warning: function Ecto.Adapters.Postgres.Connection.delete_all/1 is undefined (module Ecto.Adapters.Postgres.Connection is not available)
  lib/ecto/adapters/postgres.ex:56

warning: function Ecto.Adapters.Postgres.Connection.execute_ddl/1 is undefined (module Ecto.Adapters.Postgres.Connection is not available)
  lib/ecto/adapters/postgres.ex:56

warning: function Ecto.Adapters.Postgres.Connection.insert/4 is undefined (module Ecto.Adapters.Postgres.Connection is not available)
  lib/ecto/adapters/postgres.ex:56

warning: function Ecto.Adapters.Postgres.Connection.update/5 is undefined (module Ecto.Adapters.Postgres.Connection is not available)
  lib/ecto/adapters/postgres.ex:56

warning: function Ecto.Adapters.Postgres.Connection.update_all/1 is undefined (module Ecto.Adapters.Postgres.Connection is not available)
  lib/ecto/adapters/postgres.ex:56

Generated ecto app
==> sqlitex
Compiling 7 files (.ex)
Generated sqlitex app
==> sqlite_ecto
Compiling 7 files (.ex)
warning: the variable "args" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/sqlite_ecto/query.ex:357

warning: the variable "args" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/sqlite_ecto/query.ex:363

Generated sqlite_ecto app
==> graphql_relay
Compiling 6 files (.ex)
warning: the variable "end_offset" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/list.ex:32

warning: the variable "start_offset" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/list.ex:35

warning: the variable "end_offset" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/list.ex:36

warning: the variable "start_offset" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/list.ex:43

warning: the variable "start_offset" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/list.ex:55

warning: the variable "end_offset" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/list.ex:56

warning: the variable "query" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/ecto.ex:24

warning: the variable "query" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/ecto.ex:33

warning: the variable "query" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/ecto.ex:42

warning: the variable "query" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/ecto.ex:48

warning: the variable "query" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/ecto.ex:54

warning: the variable "query" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/ecto.ex:59

warning: the variable "has_prev_page" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/ecto.ex:83

warning: the variable "has_next_page" is unsafe as it has been set inside a case/cond/receive/if/&&/||. Please explicitly return the variable value instead. For example:

    case int do
      1 -> atom = :one
      2 -> atom = :two
    end

should be written as

    atom =
      case int do
        1 -> :one
        2 -> :two
      end

Unsafe variable found at:
  lib/graphql/relay/connection/ecto.ex:84

Generated graphql_relay app
Excluding tags: [:skip]

..................................................................................

Finished in 0.6 seconds
83 tests, 0 failures, 1 skipped

Randomized with seed 341524
```

**Comment by joshprice** [comment]
Everything within graphql-elixir/graphql-relay is fixed by #21

**Comment by joshprice** [comment]
Fixed by #21
